src/state/zustand/swapperStore/selectors.ts

```diff
@@ -92,7 +92,9 @@ export const selectCheckApprovalNeededForWallet = (
     if (!activeSwapper) throw new Error('No swapper available')
     if (!activeQuote) throw new Error('No quote available')
 
-    const { approvalNeeded } = await activeSwapper.approvalNeeded({ quote: activeQuote, wallet })
+    const { approvalNeeded } = (
+      await activeSwapper.approvalNeeded({ quote: activeQuote, wallet })
+    ).unwrap()
```

Comment:
If `approvalNeeded()` fails, can we still unwrap it like this?

Reply:
0dd508649ce33a2e64dacef3c32111c358a1bff6